### PR TITLE
add code masking

### PR DESCRIPTION
**Pull request by chlebowa**
Added private function `replace_code` to facilitate manipulating contents of `@code`.

### CODE REVIEW

**Comment by chlebowa** [comment]
Frankly, the more I look at the example, the more uncomfortable I feel exporting this function.

**Comment by chlebowa** [comment]
The function name is up for debate. I even considered `set_code` but that would imply that the whole slot can be modified.

File: R/qenv-replace_code.R [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+#'
+#' @return A modified `qenv`.
+#'
+#' @seealso [`qenv-class`], [`eval_code`]
+#'
+#' @examples
+#' q <- new_qenv()
+#' q <- eval_code(q, expression(i <- iris, ii <- subset(i, Species == "virginica")))
+#' get_code(q)
+#' q <- replace_code(q, expression(i <- iris, ii <- subset(i, Species != "virginica")))
```
Comment (gogonzo) [inline R/qenv-replace_code.R:19]: Replacing code with the same code? What do you think about this?
```suggestion
#' q <- eval_code(q, expression(print("secretpassword"))
#' get_code(q)
#' q <- replace_code(q, expression(print(askpass::askpass()))
```

Comment (chlebowa) [reply R/qenv-replace_code.R:19]: It wasn't the same though.

File: tests/testthat/test-qenv-replace_code.R [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+})
+
+testthat::test_that("non-last elements of @code are unaffected", {
+  qq <- within(qq, m <- mtcars)
+  replacement <- "i <- mtcars"
+  qr <- replace_code(qq, replacement)
+  before <- get_code(qq)[-length(get_code(qq))]
+  after <- get_code(qr)[-length(get_code(qr))]
+  testthat::expect_identical(before, after)
+})
```
Comment (gogonzo) [inline tests/testthat/test-qenv-replace_code.R:57]: Replacement of the last element of the @code is exactly what `code_replace` does and other tests checking derivative information (length, last different, non-last identical)

```suggestion
testthat::test_that("code_replace replaces last element of the code", {
  qq <- within(qq, m <- mtcars)
  replacement <- "i <- mtcars"
  qr <- replace_code(qq, replacement)
  previous <- get_code(qq)
  current <- get_code(qr)
  testthat::expect_identical(current, c(head(previous, -1), replacement)
})
```

File: R/qenv-replace_code.R [hunk 3]
```diff
@@ -0,0 +1,53 @@
+#' Replace code
+#'
+#' Replace last element of code stored in `qenv`.
+#'
+#' Replace the last element of the character vector stored in `@code` with provided `code`.
+#' This can be used to mask sensitive information, e.g. credentials.
```
Comment (pawelru) [inline R/qenv-replace_code.R:6]: When someone from the outside will be looking at this then this functionality might be super suspicious. Can you please provide few words about the context - i.e. hiding tokens, passwords etc? I know that you have this already in vignette but that someone will be probably looking only here.

Comment (chlebowa) [reply R/qenv-replace_code.R:6]: That's what worries me, though. Our intentions are rather pure and we can dress it up in pretty examples but the fact is you can put whatever you want in there.

Also, doesn't "credentials" cover what you mean?

Comment (pawelru) [reply R/qenv-replace_code.R:6]: > Also, doesn't "credentials" cover what you mean?

Yes

File: _pkgdown.yml [hunk 4]
```diff
@@ -27,6 +27,7 @@ reference:
       - concat
       - dev_suppress
       - eval_code
+      - replace_code
```
Comment (gogonzo) [inline _pkgdown.yml:30]: Internal docs shouldn't be listed in pkgdown.yml, please remove

**Review by gogonzo — APPROVED** [review]
Good to go. Please remove replace_code from pkgdown

File: _pkgdown.yml (hunk 4 above)
Comment (chlebowa) [reply _pkgdown.yml:30]: The one I always forget :roll_eyes:

**Pull request merged**